src/main/perl/FileWriter.pm

```diff
@@ -181,7 +181,14 @@ sub close
         *$self->{save} = 0;
         $str = *$self->{buf};
         *$self->{options}->{contents} = $$str;
-        $ret = LC::Check::file (*$self->{filename}, %{*$self->{options}});
+
+        my %cf_opts = %{*$self->{options}};
+        # check file is silent unless in noaction mode
+        # (or explicitly set via silent option)
+        $cf_opts{silent} = *$self->{options}->{noaction} ? 0 : 1
```

Comment:
A tertiary operator guarded by a post-hoc if() on a continuation line? Yuck! I think a full if() {} block would  be much more readable here.